### PR TITLE
feature(ajax): better elgg/Ajax handling of form data

### LINKED ISSUE #9534: Teach elgg/Ajax to handle serialized query data
When submitting forms, it is common that request data is a serialized query string returned by `$.serialize`

### LINKED ISSUE #9564: Teach elgg/Ajax to work with absolute URLs
When we render DOM elements we tend to normalize URLs and client side we use `$.attr()` to retrieve those URLs. `elgg/Ajax#path` is not really suitable unless we denormalize the URL into path and query data.

### PR DESCRIPTION
**Pull request by mrclay**
As `jQuery.serialize` is often used to set post data, this change allows a string form of `options.data` to be passed through. In this case, however, we do not trigger the request hook, as handlers expect an object.

Because we want requests to be hookable, we nudge devs toward a new `objectify` method, which "serializes" to an object so it can be passed through the hook.

All methods now accept a query string that gets passed to the underlying URL, but does not appear in the hook types. `action()` and `path()` accept absolute site URLs, as one might get from `$anchor.prop('href')`.

Fixes #9534
Fixes #9564

### CODE REVIEW

**Comment by mrclay** [comment]
@hypeJunction please take a look. This adds complete tests and allows giving URLs for action, path, and form (action URL).

File: views/default/elgg/Ajax.js [hunk 1]
```diff
@@ -124,17 +145,26 @@ define(function (require) {
...
-		 * @param {String} path    URL path. e.g. "foo/bar"
+		 * @param {String} path    path or URL. e.g. "foo/bar"
 		 * @param {Object} options {@link jQuery#ajax}. See fetch()
 		 * @returns {jqXHR}
 		 */
 		this.path = function (path, options) {
 			elgg.assertTypeOf('string', path);
 
+			// https://example.org/elgg/foo/?arg=1#bar => foo/?arg=1
+			if (path.indexOf(site_url) === 0) {
```
Comment (hypeJunction) [inline views/default/elgg/Ajax.js]: I think we should bail if it's an external URL. I don't think we should encourage use of this API for fetching external resources.

**Comment by hypeJunction** [comment]
There is a lot to process. I thibk it looks good and needs testing in battle. If you are confident enough, let's merge and I will apply it to discussions. I have also strated updating hypeLists to use this.
One other thing that comes to mind is supporting jsonp to unwrap data, but that's something for another ticket (and probably only makes sense for external URLs)

File: views/default/elgg/Ajax.js (hunk 1 above)
Comment (mrclay) [inline views/default/elgg/Ajax.js]: Added test/code for this

**Comment by mrclay** [comment]
I removed the new PHP function and added tests for the system messages and deps handling.
